components/engine_rocks/src/engine.rs

```diff
@@ -187,6 +194,72 @@ impl SyncMutable for RocksEngine {
     }
 }
 
+pub struct DummyRocksEngineFactory {
+    pub engine: Option<RocksEngine>,
+    pub root_path: String,
+}
+
+impl TabletFactory<RocksEngine> for DummyRocksEngineFactory {
+    fn create_tablet(&self, _id: u64, _suffix: u64) -> Result<RocksEngine> {
+        Ok(self.engine.as_ref().unwrap().clone())
+    }
+    fn open_tablet_raw(&self, _path: &Path, _readonly: bool) -> Result<RocksEngine> {
+        Ok(self.engine.as_ref().unwrap().clone())
+    }
+    fn create_shared_db(&self) -> Result<RocksEngine> {
+        Ok(self.engine.as_ref().unwrap().clone())
+    }
+    fn destroy_tablet(&self, _id: u64, _suffix: u64) -> Result<()> {
+        Ok(())
+    }
+    fn exists_raw(&self, _path: &Path) -> bool {
+        true
+    }
+    fn tablet_path(&self, _id: u64, _suffix: u64) -> PathBuf {
+        PathBuf::from(&self.root_path)
+    }
+    fn tablets_path(&self) -> PathBuf {
+        PathBuf::from(&self.root_path)
+    }
+
+    fn clone(&self) -> Box<dyn TabletFactory<RocksEngine> + Send> {
+        if self.engine.is_none() {
+            return Box::<DummyRocksEngineFactory>::new(DummyRocksEngineFactory {
+                engine: None,
+                root_path: self.root_path.clone(),
+            });
+        }
+        Box::<DummyRocksEngineFactory>::new(DummyRocksEngineFactory {
+            engine: Some(self.engine.as_ref().unwrap().clone()),
+            root_path: self.root_path.clone(),
+        })
+    }
+
+    fn loop_tablet_cache(&self, mut f: Box<dyn FnMut(u64, u64, &RocksEngine) + '_>) {
+        if let Some(engine) = &self.engine {
+            f(0, 0, engine);
+        }
+    }
+
+    fn set_shared_db_block_cache_size(&self, cf: &str, size: ReadableSize) -> Result<()> {
+        let opt = self.engine.as_ref().unwrap().get_options_cf(cf)?;
+        opt.set_block_cache_capacity(size.0)
```

Comment:
Seems you don't set the opt back to the engine?

Reply:
it's the ref and no need to setback.